Add dedicated dbt_cloud.yml reference page (#9177)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt_cloud.yml.md

---
title: "dbt_cloud.yml file"
id: dbt_cloud.yml
sidebar_label: "dbt_cloud.yml"
description: "Reference for the dbt_cloud.yml credentials file used by the dbt CLI and dbt VS Code extension to connect to the dbt platform."
pagination_next: null
---

> The `dbt_cloud.yml` file stores the credentials that dbt tools &mdash; like the <Constant name="platform_cli" />, the [dbt VS Code extension](/docs/about-dbt-extension), and more &mdash; use to authenticate with <Constant name="dbt_platform" />. You can download it from <Constant name="dbt_platform" /> and save it locally to your `.dbt` directory.

Author: ```suggestion
The `dbt_cloud.yml` file stores the credentials that dbt tools &mdash; like the [<Constant name="platform_cli" />](/docs/platform/dbt-cli-installation), the [dbt VS Code extension](/docs/about-dbt-extension), and more &mdash; use to authenticate with <Constant name="dbt_platform" />. You can download it from <Constant name="dbt_platform" /> and save it locally to your `.dbt` directory.
```

This page covers:

> - [How to download it and set up the `.dbt` directory](#download-dbt_cloudyml) for the dbt CLI or the VS Code extension

Reviewer: ```suggestion
- [How to download it and set up the `.dbt` directory](#download-dbt_cloudyml) for the <Constant name="platform_cli" /> or the VS Code extension
```

- [The file structure](#file-structure) and field reference
- [The companion `dbt-cloud` block](#the-dbt-cloud-block-in-dbt_projectyml) in `dbt_project.yml`

:::tip Keep this file safe
The `dbt_cloud.yml` file contains API keys. Store it securely and make sure you _do not_ commit it to version control.
:::

## Download dbt_cloud.yml
import DbtDirectoryFaq from '/snippets/_dbt-directory-faq.md';

> How you download the file depends on whether you're configuring the <Constant name="platform_cli" /> or the [dbt VS Code extension](/docs/about-dbt-extension).

Reviewer: Maybe link out dbt CLI as dbt VS Code is linked out?

> How you download the file depends on whether you're configuring the <Constant name="platform_cli" /> or the [dbt VS Code extension](/docs/about-dbt-extension).

Author: ```suggestion
How you download the file depends on whether you're configuring the [<Constant name="platform_cli" />](/docs/platform/dbt-cli-installation) or the [dbt VS Code extension](/docs/about-dbt-extension).
```

1. In <Constant name="dbt_platform" />, select the project you want to work on. The project must already have a [development environment](/docs/dbt-platform-environments#create-a-development-environment) set up.
2. Go to **Account settings** → **Your profile**, then follow the steps for your tool:

>   - dbt CLI: Navigate to **CLI** → **Configure Cloud authentication** and click **Download CLI configuration file**.

Reviewer: ```suggestion
  - <Constant name="platform_cli" />: Navigate to **CLI** → **Configure Cloud authentication** and click **Download CLI configuration file**.
```

  - dbt VS Code extension: Navigate to **VS Code Extension** → **Set up your credentials** and click **Download credentials**.
    The downloaded `dbt_cloud.yml` includes your [Personal access token (PAT)](/docs/dbt-apis/user-tokens).
3. Move the file to your `.dbt` directory. If you don't have one yet, try one of these quick setup (one command) options:
    <Tabs>
    <TabItem value="mac-linux" label="Mac / Linux">

    ```bash
    mkdir -p ~/.dbt && mv ~/Downloads/dbt_cloud.yml ~/.dbt/dbt_cloud.yml
    ```

    </TabItem>
    <TabItem value="windows" label="Windows">

    ```powershell

>     mkdir %USERPROFILE%\.dbt 2>nul & move %USERPROFILE%\Downloads\dbt_cloud.yml %USERPROFILE%\.dbt\dbt_cloud.yml

Reviewer: Hiya Mirna

This command mkdir %USERPROFILE%\.dbt appears to be a windows command but it appears under the the macOS instructions which might confuse readers

4. Confirm that the `project-id` in your [`dbt_project.yml` `dbt-cloud` block](#the-dbt-cloud-block-in-dbt_projectyml) matches the project you're working on. This registers and connects your tool to <Constant name="dbt_platform" /> and enables platform features such as <Constant name="mesh" /> and deferral.

## File structure

Your `dbt_cloud.yml` file has the following structure:

<File name="dbt_cloud.yml">

```yaml
version: "1"
context:
  active-host: your_active_host       # for example, "abc123.us1.dbt.com"
  active-project: your_project_id     # for example, "123456"
projects:
  - project-name: your_project_name
    project-id: your_project_id
    account-name: your_account_name
    account-id: your_account_id
    account-host: your_account_host   # for example, "abc123.us1.dbt.com"
    token-name: your_token_name       # for example, "cloud-cli-1234"
    token-value: your_token_value

  - project-name: your_project_name
    project-id: your_project_id
    account-name: your_account_name
    account-id: your_account_id
    account-host: your_account_host
    token-name: your_token_name
    token-value: your_token_value
```

</File>

> For exmaple, if I had a Jaffle and wizard shop account, here's what my `dbt_cloud.yml` file would look like:

Reviewer: ```suggestion
For example, if I had a Jaffle and wizard shop account, here's what my `dbt_cloud.yml` file would look like:
```

Author: great catch!

> For exmaple, if I had a Jaffle and wizard shop account, here's what my `dbt_cloud.yml` file would look like:

Reviewer: Following the style guide, maybe amend this to second person?

> For exmaple, if I had a Jaffle and wizard shop account, here's what my `dbt_cloud.yml` file would look like:

Author: ```suggestion
For exmaple, if you had a Jaffle and wizard shop account, here's what my `dbt_cloud.yml` file would look like:
```

| `projects.account-host` | Yes | The host for your account, for example `cloud.getdbt.com`, `emea.dbt.com`, or your single-tenant access URL. |
| `projects.token-name` | Yes | A name for the [Personal access token (PAT)](/docs/dbt-apis/user-tokens). |
| `projects.token-value` | Yes | The PAT value. Treat this as a secret. |

## The dbt-cloud block in dbt_project.yml

The `dbt-cloud` block is a companion config that lives in your project's `dbt_project.yml` file (not in `dbt_cloud.yml`). It tells the <Constant name="platform_cli" />, the <Constant name="studio_ide" />, and <Constant name="fusion" /> which <Constant name="dbt_platform" /> project your local project corresponds to.

<File name="dbt_project.yml">

```yaml
name:
version:
# Your project configs...

dbt-cloud:
  project-id: your_project_id
  defer-env-id: '123456'  # optional
```

</File>

| Field | Required | Description |
|-------|----------|-------------|
| `project-id` | Yes | The <Constant name="dbt_platform" /> project ID this local project maps to. Find it in the URL when viewing your project (for example, `https://YOUR_ACCESS_URL/develop/26228/projects/123456` → `123456`). |

> | `defer-env-id` | No | The environment ID to defer to for build artifacts. Used for <Constant name="fusion" /> [auto-deferral](/docs/platform/about-defer) and <Constant name="platform_cli" /> deferral overrides. |

Reviewer: Hiya Mirna

Given this page is the central reference for `dbt_cloud.yml`, it might benefit from a related content section
